File: src/collatz.cs
class Collatz
{
    public static int collatz(int n)
    {
        System.Console.WriteLine(n);
        if (n == 1)
            return 1;
        if (n % 2 == 1)
            return collatz(3 * n + 1);
        return collatz(n / 2);
    }

    public static void Main(string[] args)
    {
        collatz(9);
    }
}

File: src/collatz.scala
object Collatz {
    def collatz(x: Int): Int = {
        println(x)
        if (x <= 1) 1
        else if (x % 2 == 0) collatz(x / 2)
        else collatz(3*x+1)
        }

    def main(args: Array[String]) {
        collatz(9)
    }
}
